### PR TITLE
Add a simple unit test to confirm that the r executable actually works.

### PR DESCRIPTION
**Pull request by jeroenooms**
Add a simple function that invokes little r from within big R. Function is not actually exported currently, but could be if that is useful.

### CODE REVIEW

**Comment by eddelbuettel** [comment]
Very much in  favour of tests but we have an existing directory full of them -- and I fail to see what printing `Sys.info()` does for me, particularly when the result is ignored!!

But I like the idea of `littler:::test()`. So I may merge this after all...

**Comment by jeroenooms** [comment]
But your existing tests do not get triggered during `CMD check`? It's not so much about printing sys.info, it is that `test()` will raise an error if it cannot find or execute littler.

**Comment by eddelbuettel** [comment]
Please remain aware of the fact that up until Sunday the package had a happy nine-year existence as a standard tarball of the `configure; make; make install` variety.   It's has been an R package for a few hours.  So patience grasshopper :)

So you want to look into how to tie some of the `inst/script-tests/` files into `R CMD check ...` tests?

**Comment by jeroenooms** [comment]
I think the current test does everything you need. It simply verifies that the `r` executable is where we expect it to be and that it is executable, without assuming any particular shell or platform. I think this is something that should be part of the automatic `CMD check` on any platform.

Your other test scripts assume `bash` or shebang or presence of a `|` pipe operator which might not be present on every system.

**Comment by eddelbuettel** [comment]
I politely disagree.  If I run a test, I check an outcome.  This PR does not do that.

**Comment by jeroenooms** [comment]
Fair enough, I added this to the test.

File: R/test.R [hunk 1]
```diff
@@ -0,0 +1,4 @@
+test <- function(src = "print(sessionInfo())"){
+  r <- file.path(system.file(package="littler"), "bin/r")
```
Comment (eddelbuettel) [inline R/test.R]: With what we have in `src/Makevars.in`, should this be more generic than just `r`?  I appreciate the email note by @s-u reminding us that there is only OS X binary too.  So maybe I could simplify `src/Makevars.in`?   OTOH we may also get some brave person submitting build fixes for Windoze just how it happened for RInside a few years ago (and the two packages share a fair amount of structure around embedding R).

**Comment by eddelbuettel** [comment]
I'll fold it in and just generalize that `bin/r` bit.  Over time, someone may come and get us Windows builds given that it also works for RInside...

**Comment by eddelbuettel** [comment]
This promptly went belly-up in the initial attempt to upload littler 0.3.0.  From mail from CRAN:

```
* checking tests ...
  ERROR
Running the tests in ‘tests/simpleTests.R’ failed.
Last 13 lines of output:
   Type 'q()' to quit R.

   > library(littler)
   The littler package provides 'r' as a binary.
   See 'vignette("littler-examples") for some illustrations.
   On OS X, 'r' and 'R' are the same so 'lr' is an alternate name for 
littler.
   You could link to the 'r' binary installed in
   '/Users/murdoch/R/incoming/littler.Rcheck/littler/bin/r'
   as '/usr/local/bin/lr' in order to use 'lr' for scripting.
   > stopifnot(identical(littler:::test(), "4"))   # default test is 
cat(2+2)
   sh: line 1: 27007 Trace/BPT trap: 5 
'/Users/murdoch/R/incoming/littler.Rcheck/littler/bin/r' -e 'cat(2 + 2)' 
2>&1
   Error: identical(littler:::test(), "4") is not TRUE
   In addition: Warning message:
   running command 
''/Users/murdoch/R/incoming/littler.Rcheck/littler/bin/r' -e 'cat(2 + 
2)' 2>&1' had status 133
   Execution halted
```

I responded by making this conditional on Linux, see [the commit](https://github.com/eddelbuettel/littler/commit/190bc9ff319a0f441d48c1d31313ee5ec0313d6e). 

If you feel you'd want to fix this, I'd welcome a closer look given that I don't have an OS X account.  If you don't care I won't worry either.  It did make it to CRAN with the fix.

**Pull request merged**